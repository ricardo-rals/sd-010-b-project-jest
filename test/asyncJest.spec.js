// const assert = require('assert');
const answerPhone = require('../src/asyncJest');
/*
A função answerPhone recebe um parâmetro boleano.
Dependendo do parâmetro o retorno da função varia, veja a função no arquivo 'src/asyncJest.js'

Complete o código abaixo para testar as situações em que
a função recebe como parâmetro true e false, respectivamente.

ATENÇÃO!!! Edite apenas este arquivo. Não altere os arquivos da pasta 'src'.
*/

describe('o retorno do telefonema', () => {
  test('atende', () => {
    expect.assertions(1);
    return answerPhone(true).then((response) => {
      expect(response).toEqual('Oi!');
    });
  });
  test('ocupado', () =>
    expect(answerPhone(false)).rejects.toEqual(
      Error('Infelizmente não podemos atender...'),
    ));
});

// Pra testar, mudei detalhes das respostas e o teste não passou. Por isso acredito que esteja correto.

// expect(() => answerPhone(false).toThrow('Infelizmente não podemos atender...'))
