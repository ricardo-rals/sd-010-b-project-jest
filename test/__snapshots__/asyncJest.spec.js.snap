// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`o retorno do telefonema ocupado 1`] = `"received is not a function"`;
